src/PHPHtmlParser/Selector.php

```diff
@@ -168,7 +169,9 @@ protected function parseSelectorString($selector)
     protected function seek(array $nodes, array $rule, array $options)
     {
         // XPath index
-        if (count($rule['tag']) > 0 &&
+        if ($rule['tag'] instanceof Countable && 
+            count($rule['tag']) > 0 &&
+            $rule['key'] instanceof Countable &&
```

Comment:
`$rule['key']` is compared with an int below, so this check is incorrect as well. Since it's supposed to be an index, I'd suggest to check `$rule['key'] >  0`